src/scalems/radical/raptor.py

```diff
@@ -829,11 +1313,287 @@ def worker_description(*,
+    """
+
+
+class _RaptorTaskDescription(typing.TypedDict):
+    """Describe a Task to be executed through a Raptor Worker.
+
+    A specialization of `radical.pilot.TaskDescription`.
+
+    Note the distinctions of a TaskDescription to be processed by a raptor.Master.
+
+    The meaning or utility of some fields is dependent on the values of other fields.
+
+    TODO: We need some additional fields, like *environment* and fields related
+     to launch method and resources reservation. Is the schema for rp.TaskDescription
+     sufficiently strong and well-documented that we can remove this hinting type?
+     (Check again at RP >= 1.19)
+    """
+    uid: str
+    """Unique identifier for the Task across the Session."""
+
+    executable: str
+    """Unused by Raptor tasks."""
+
+    scheduler: str
+    """The UID of the raptor.Master scheduler task.
+
+    This field is relevant to tasks routed from client TaskManagers. It is not
+    used for tasks originating in master tasks.
+
+    .. ref https://github.com/SCALE-MS/scale-ms/discussions/258#discussioncomment-4087870
+
+    """
+
+    metadata: Encodable
+    """An arbitrary user-provided payload.
+
+    May be any type that is encodable by :py:mod:`msgpack` (i.e. built-in types).
+    """
+
+    mode: str
+    """The executor mode for the Worker to use.
+
+    For ``rp.TASK_FUNCTION`` mode, either *function* or *method* must name a task executor.
+    Depending on the Worker (sub)class, resources such as an `mpi4py.MPI.Comm`
+    will be provided as the first positional argument to the executor.
+    ``*args`` and ``**kwargs`` will be provided to the executor from the corresponding
+    fields.
+
+    See Also:
+        * :py:data:`CPI_MESSAGE`
+        * :py:class:`RaptorTaskExecutor`
+    """
+
+    function: str
+    """Executor for ``rp.TASK_FUNCTION`` mode.
+
+    Names the callable for dispatching.
+
+    The callable can either be a function object present in the namespace of the interpreter launched
+    by the Worker for the task, or a `radical.pilot.pytask` pickled function object.
+    """
+
+    args: list
+    """For ``rp.TASK_FUNCTION`` mode, list of positional arguments provided to the executor function."""
+
+    kwargs: dict
+    """For ``rp.TASK_FUNCTION`` mode, a dictionary of key word arguments provided to the executor function."""
+
+
+class TaskDictionary(typing.TypedDict):
+    """Task representations seen by *request_cb* and *result_cb*.
+
+    Other fields may be present, but the objects in the sequences provided to
+    :py:meth:`scalems.radical.raptor.ScaleMSMaster.request_cb()` and
+    :py:meth:`scalems.radical.raptor.ScaleMSMaster.result_cb()` have the following fields.
+    Result fields will not be populated until the Task runs.
+
+    For the expected fields, see the source code for
+    :py:meth:`~radical.pilot.Task.as_dict()`:
+    https://radicalpilot.readthedocs.io/en/stable/_modules/radical/pilot/task.html#Task.as_dict
```

Comment:
Is this a public and documented data structure yet?

Reply:
No, not yet.  It will be eventually, but the timeline for that is somewhat unclear.

Reply:
Thanks. I'll expose this in the executor_radical.rst, then, for the foreseeable future.

It looks like we will have to bump the required RP version again to 1.19.0 to simplify compatibility. Do we have an expected release date for that?